source/type/headsign_handler.cpp

```diff
@@ -0,0 +1,169 @@
+/* Copyright © 2001-2015, Canal TP and/or its affiliates. All rights reserved.
+
+This file is part of Navitia,
+    the software to build cool stuff with public transport.
+
+Hope you'll enjoy and contribute to this project,
+    powered by Canal TP (www.canaltp.fr).
+Help us simplify mobility and open public transport:
+    a non ending quest to the responsive locomotion way of traveling!
+
+LICENCE: This program is free software; you can redistribute it and/or modify
+it under the terms of the GNU Affero General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU Affero General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License
+along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+Stay tuned using
+twitter @navitia
+IRC #navitia on freenode
+https://groups.google.com/d/forum/navitia
+www.navitia.io
+*/
+
+#include "headsign_handler.h"
+#include "utils/functions.h"
+
+namespace navitia { namespace type {
+
+void HeadsignHandler::change_name_and_register_as_headsign(VehicleJourney& vj,
+                                                           const std::string& new_name) {
+    std::string prev_name = vj.name;
+    vj.name = new_name;
+    headsign_mvj[vj.name].insert(vj.meta_vj);
```

Comment:
are you sure `vj.name` is always a valid key of `headsign_mvj`? Pay attention when using `[]` directly :)